frontend/src/components/ViewTrips/trip-utils.js

```diff
@@ -31,52 +36,109 @@ export function getCleanedTextInput(rawInput, defaultValue) {
     return email !== '';
   });
 
   return await authUtils.getUserUidArrFromUserEmailArr(cleanedCollaboratorEmailArr);
 }
 
+/**
+ * Returns the new `accepted_collaborator_uid_arr` field value for the updated
+ * trip document.
+ *
+ * @param {!string[]} formCollabUidArr Array of the collaborator uids
+ *     corresponding to the emails in the `collaborator_email_arr` field of the
+ *     {@link RawTripData} instance coming from the `SaveTripModal` component.
+ * @param {!string[]} curAccepCollabUidArr Array stored in the field
+ *     `accepted_collaborator_uid_arr` of the {@link TripData} obj
+ *     `curTripData`.
+ * @return {!string[]} Array containing the set intersection of two params
+ *     `formCollabUidArr` and `curAccepCollabUidArr` converted to sets.
+ */
+export function getCurAcceptedCollabUidArr(formCollabUidArr, curAccepCollabUidArr) {
+  const formCollabUidSet = new Set(formCollabUidArr);
+  const curAccepCollabUidSet = new Set(curAccepCollabUidArr);
+  return [...formCollabUidSet].filter(el => curAccepCollabUidSet.has(el));
+}
+
+/**
+ * Returns the new `pending_collaborator_uid_arr` field value for the updated
+ * trip document.
+ *
+ * @param {!string[]} formCollabUidArr Array of the collaborator uids
+ *     corresponding to the emails in the `collaborator_email_arr` field of the
+ *     {@link RawTripData} instance coming from the `SaveTripModal` component.
+ * @param {!string[]} curAccepCollabUidArr Array stored in the field
+ *     `accepted_collaborator_uid_arr` of the {@link TripData} obj
+ *     `curTripData`.
+ * @return {!string[]} Array containing the set difference of two params
+ *     `formCollabUidArr` and `curAccepCollabUidArr` converted to sets.
+ */
+export function getCurPendingCollabUidArr(formCollabUidArr, curAccepCollabUidArr) {
+  const formCollabUidSet = new Set(formCollabUidArr);
+  const curAccepCollabUidSet = new Set(curAccepCollabUidArr);
+  return [...formCollabUidSet].filter(el => !curAccepCollabUidSet.has(el));
+}
+
 /**
  * Returns a promise containing the formatted and cleaned {@link RawTripData}
  * that will be used to instantiate the the created trip document.
  *
- * We know that {@link RawTripData} will contain all of the necessary fields for
- * a trip document (except updated timestamp) because each key-value pair is
- * explicitly included. This means, only the value corresponding to each key
- * needs to be checked.
+ * Formatting done to convert {@link RawTripData} to {@link TripData}:
+ * - update timestamp: get current time with `Firestore.Timestamp.now()`.
+ * - plain text fields: basic input cleaning by providing default values.
+ * - date fields: ISO strings are converted to `Firestore.Timestamp` objs.
+ * - collaborator fields: Determine collab uid arrays based on `curTripData`.
+ *   New trips will add the current user to the accepted list and all others
+ *   to pending. Existing trips use `getCurAcceptedCollabUidArr` and
```

Comment:
Accepted is not consistent with the "Accep" above. Make them consistent?

Reply:
See above. Fixed.